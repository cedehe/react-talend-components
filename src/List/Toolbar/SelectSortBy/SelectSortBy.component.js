import React, { PropTypes } from 'react';
import { Button, Nav, NavDropdown, MenuItem } from 'react-bootstrap';
import uuid from 'uuid';

import theme from './SelectSortBy.scss';

/**
 * @param {object} props react props
 * @example
 <SelectSortBy name="Hello world"></SelectSortBy>
 */
function SelectSortBy(props) {
	let selected;
	props.sortBy.forEach((sortBy) => {
		if (sortBy.selected) {
			selected = sortBy;
		}
	});
	let onSelectSortBy = props.onSelectSortBy;
	if (onSelectSortBy) {
		onSelectSortBy = (key, event) => props.onSelectSortBy(
			event,
			{
				sortBy: key.id,
				sortDesc: !!props.sortDesc,
			}
		);
	}
	const toggleSortOrder = () => {
		if (props.onSelectSortBy) {
			props.onSelectSortBy(
				event,
				{
					sortBy: selected.id,
					sortDesc: !props.sortDesc,
				}
			);
		}
	};
	const sortById = props.id && `${props.id}-sort-by`;
	return (
		<Nav className={theme['tc-list-toolbar-sort-by']}>
			<label className="navbar-text" htmlFor={sortById}>Sort by:</label>
			<NavDropdown
				id={sortById || uuid.v4()}
				title={selected ? selected.name : 'N.C'}
				onSelect={onSelectSortBy}
			>
				{props.sortBy.map((sortBy, index) => (
					<MenuItem
						id={props.id && `${props.id}-sort-by-item-${sortBy.name}`}
						key={index}
						eventKey={sortBy}
					>
						{sortBy.name}
					</MenuItem>
				))}
			</NavDropdown>
			<Button
				id={props.id && `${props.id}-sort-order`}
				className="navbar-btn btn"
				bsStyle="link"
				role="button"
				onClick={toggleSortOrder}
			>
				{props.sortDesc ? 'DESCENDING' : 'ASCENDING'}
			</Button>
		</Nav>
	);
}

SelectSortBy.propTypes = {
	id: PropTypes.string,
	onSelectSortBy: PropTypes.func,
	sortBy: PropTypes.arrayOf(
		PropTypes.shape({
			id: PropTypes.string,
			name: PropTypes.string,
			selected: PropTypes.bool,
		})
	),
	sortDesc: PropTypes.bool,
};

export default SelectSortBy;
