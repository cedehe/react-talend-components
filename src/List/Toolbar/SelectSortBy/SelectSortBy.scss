// SelectSortBy styles

$tc-list-toolbar-display-color: $dove-gray !default;

.tc-list-toolbar-sort-by {
	:global(.navbar-btn.btn) {
		color: $tc-list-toolbar-display-color;
		font-size: 1em;
		margin: 0;
		padding-top: $navbar-padding-vertical;
		padding-bottom: $navbar-padding-vertical;
		letter-spacing: 0;
	}
}
