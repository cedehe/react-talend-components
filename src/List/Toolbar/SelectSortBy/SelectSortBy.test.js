import React from 'react';
import renderer from 'react-test-renderer';

import SelectSortBy from './SelectSortBy.component';

jest.mock('react-dom');

describe('SelectSortBy', () => {
	it('should render', () => {
		// given
		const props = {
			onSelectSortBy: jest.fn(),
			sortBy: [
				{ id: 'id', name: 'Name', selected: true },
				{ id: 'name', name: 'Name' },
			],
		};

		// when
		const wrapper = renderer.create(
			<SelectSortBy {...props} />
		).toJSON();

		// then
		expect(wrapper).toMatchSnapshot();
	});

	it('should render without sortBy selected', () => {
		// given
		const props = {
			onSelectSortBy: jest.fn(),
			sortBy: [
				{ id: 'id', name: 'Name' },
				{ id: 'label', name: 'Label' },
			],
		};

		// when
		const wrapper = renderer.create(
			<SelectSortBy {...props} />
		).toJSON();

		// then
		expect(wrapper).toMatchSnapshot();
	});


	it('should render id if provided', () => {
		// given
		const props = {
			id: 'toolbar',
			onSelectSortBy: jest.fn(),
			sortBy: [
				{ id: 'id', name: 'Name', selected: true },
				{ id: 'name', name: 'Name' },
			],
		};

		// when
		const wrapper = renderer.create(
			<SelectSortBy {...props} />
		).toJSON();

		// then
		expect(wrapper).toMatchSnapshot();
	});
});
