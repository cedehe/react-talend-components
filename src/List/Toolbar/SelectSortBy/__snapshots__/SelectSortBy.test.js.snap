exports[`SelectSortBy should render 1`] = `
<ul
  className="nav"
  role={null}>
  <label
    active={undefined}
    activeHref={undefined}
    activeKey={undefined}
    className="navbar-text"
    htmlFor={undefined}
    onSelect={null}>
    Sort by:
  </label>
  <li
    className="dropdown"
    style={undefined}>
    <a
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle"
      href=""
      id={42}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button">
      Name
       
      <span
        className="caret" />
    </a>
    <ul
      aria-labelledby={42}
      className="dropdown-menu"
      role="menu">
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          id={undefined}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          Name
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          id={undefined}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          Name
        </a>
      </li>
    </ul>
  </li>
  <button
    activeHref={undefined}
    activeKey={undefined}
    className="navbar-btn btn btn btn-link"
    disabled={false}
    id={undefined}
    onClick={[Function]}
    onSelect={null}
    role="button"
    type="button">
    ASCENDING
  </button>
</ul>
`;

exports[`SelectSortBy should render id if provided 1`] = `
<ul
  className="nav"
  role={null}>
  <label
    active={undefined}
    activeHref={undefined}
    activeKey={undefined}
    className="navbar-text"
    htmlFor="toolbar-sort-by"
    onSelect={null}>
    Sort by:
  </label>
  <li
    className="dropdown"
    style={undefined}>
    <a
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle"
      href=""
      id="toolbar-sort-by"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button">
      Name
       
      <span
        className="caret" />
    </a>
    <ul
      aria-labelledby="toolbar-sort-by"
      className="dropdown-menu"
      role="menu">
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          id="toolbar-sort-by-item-Name"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          Name
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          id="toolbar-sort-by-item-Name"
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          Name
        </a>
      </li>
    </ul>
  </li>
  <button
    activeHref={undefined}
    activeKey={undefined}
    className="navbar-btn btn btn btn-link"
    disabled={false}
    id="toolbar-sort-order"
    onClick={[Function]}
    onSelect={null}
    role="button"
    type="button">
    ASCENDING
  </button>
</ul>
`;

exports[`SelectSortBy should render without sortBy selected 1`] = `
<ul
  className="nav"
  role={null}>
  <label
    active={undefined}
    activeHref={undefined}
    activeKey={undefined}
    className="navbar-text"
    htmlFor={undefined}
    onSelect={null}>
    Sort by:
  </label>
  <li
    className="dropdown"
    style={undefined}>
    <a
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle"
      href=""
      id={42}
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button">
      N.C
       
      <span
        className="caret" />
    </a>
    <ul
      aria-labelledby={42}
      className="dropdown-menu"
      role="menu">
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          id={undefined}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          Name
        </a>
      </li>
      <li
        className=""
        role="presentation"
        style={undefined}>
        <a
          href=""
          id={undefined}
          onClick={[Function]}
          onKeyDown={[Function]}
          role="menuitem"
          tabIndex="-1">
          Label
        </a>
      </li>
    </ul>
  </li>
  <button
    activeHref={undefined}
    activeKey={undefined}
    className="navbar-btn btn btn btn-link"
    disabled={false}
    id={undefined}
    onClick={[Function]}
    onSelect={null}
    role="button"
    type="button">
    ASCENDING
  </button>
</ul>
`;
